packages/core/contracts/cross-chain-oracle/chain-adapters/ParentMessengerBase.sol

```diff
@@ -0,0 +1,80 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+contract ParentMessengerBase is Ownable {
+    uint256 public chainId;
+
+    address public childMessenger;
+
+    address public oracleHub;
+    address public governorHub;
+
+    address public oracleSpoke;
+    address public governorSpoke;
+
+    uint32 public defaultGasLimit = 5_000_000;
+
+    modifier onlyPrivilegedCaller() {
+        require(msg.sender == oracleHub || msg.sender == governorHub, "Only privileged caller");
+        _;
+    }
+
+    /**
+     * @notice Construct the ParentMessengerBase contract.
+     * @param _chainId The chain id of the L2 network this messenger should connect to.
+     **/
+    constructor(uint256 _chainId) {
+        chainId = _chainId;
```

Comment:
Why not read from block.chainId like `OracleSpoke._stampAncillaryData` does?

Reply:
Oh I see. I think we should rename this to `childChainId` to be more clear.

Reply:
sure. done